controller/api/destination/watcher/endpoints_watcher_test.go

```diff
@@ -38,46 +41,94 @@ func addressString(address Address) string {
 	return addressString
 }
 
+func (bel *bufferingEndpointListener) Added() []string {
+	bel.Lock()
+	defer bel.Unlock()
+	return bel.added
```

Comment:
Aren't slices passed by reference in go?  if so, I think this slice is still shared and can still encounter a data race.

Reply:
@adleong I think the last commit should fix that.